### PR TITLE
Upgrade mkdirp dependency, remove node8 support

### LINKED ISSUE #2544: Upgrade mkdirp to latest version, drop node8 support.
Node8 is not supported anymore, this is now the time to drop it here too by upgrading to the v1 branch of mkdirp (which requires node10)

### PR DESCRIPTION
**Pull request by rejas**
Fixes #2544

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,6 +95,12 @@
     "update-changelog": "auto-changelog --commit-limit false --package",
     "update-license": "node scripts/generate-license.js && git add LICENSE"
   },
+  "directories": {
+    "test": "test"
```
Comment (patrickkettner) [inline package.json:99]: any reason?

Comment (rejas) [reply package.json:99]: just alphabetical order. for what tool is that useful bytheway?

Comment (patrickkettner) [reply package.json:99]: oh, I didn't realize this was here in the first place.

looks like it was added for grunt back in the day. it did what [you did here](https://github.com/Modernizr/Modernizr/blob/master/gulpfile.babel.js#L16-L30)

id say chuck it, while we are here

Comment (rejas) [reply package.json]: chucked it!

Comment (patrickkettner) [reply package.json]: image

**Comment by patrickkettner** [comment]
:shipit:

**Pull request merged**